xds/internal/xdsclient/xdsresource/unmarshal_cds.go

```diff
@@ -81,6 +82,28 @@ const (
 )
 
 func validateClusterAndConstructClusterUpdate(cluster *v3clusterpb.Cluster) (ClusterUpdate, error) {
+	stringMD := make(map[string]string)
+	if fmd := cluster.GetMetadata().GetFilterMetadata(); fmd != nil {
+		if val, ok := fmd["com.google.csm.telemetry_labels"]; ok {
+			/*
+				"service_name" = "",
+				"service_namespace" = ""
+			*/
+			if fields := val.GetFields(); fields != nil {
+				if val, ok := fields["service_name"]; ok {
+					if _, isStringVal := val.GetKind().(*structpb.Value_StringValue); isStringVal {
+						stringMD["service_name"] = val.GetStringValue()
```

Comment:
This feels weird.  We check the type of a thing, then call `GetStringValue` on some _other_ thing.  Does `GetStringValue` do something different if the `Kind()` is not `StringValue`?

Reply:
Yeah, this is the best solution that I could come up with. GetStringValue() returns a string, not a *string, so there is no way to distinguish between an empty string (a valid label value, which will be recorded) and an unset string (should eventually record method labels as "unknown" if unset). Thus, I don't set if not a string value (not set = "unknown" in OpenTelemetry component) and set it even if it's empty if it's present.

Reply:
I _believe_ you can do:

```go
if s, ok := val.AsInterface().(string); ok {
	stringMD["service_name"] = s
}
```

But I'm not sure that that's much better.